common/common-http/src/main/java/org/ow2/proactive/http/HttpClientBuilder.java

```diff
@@ -0,0 +1,190 @@
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
+ * USA
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ *
+ *  Initial developer(s):               The ProActive Team
+ *                        http://proactive.inria.fr/team_members.htm
+ *  Contributor(s):
+ *
+ * ################################################################
+ * $$PROACTIVE_INITIAL_DEV$$
+ */
+package org.ow2.proactive.http;
+
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.net.ssl.SSLContext;
+
+import org.ow2.proactive.web.WebProperties;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLContextBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+
+public class HttpClientBuilder {
+
+    private static final AllowAllTrustStrategy ACCEPT_ANY_CERTIFICATE_TRUST_STRATEGY = new AllowAllTrustStrategy();
+
+    private Boolean allowAnyHostname;
+
+    private Boolean allowAnyCertificate;
+
+    private int maxConnections;
+
+    private RequestConfig requestConfig;
+
+    private boolean useContentCompression;
+
+    private boolean useSystemProperties;
+
+    public HttpClientBuilder() {
+        this.useContentCompression = true;
+    }
+
+
+    public HttpClientBuilder allowAnyHostname(boolean acceptAnyHostname) {
+        this.allowAnyHostname = acceptAnyHostname;
+        return this;
+    }
+
+    public HttpClientBuilder allowAnyCertificate(boolean acceptAnyCertificate) {
+        this.allowAnyCertificate = acceptAnyCertificate;
+        return this;
+    }
+
+    public HttpClientBuilder disableContentCompression() {
+        this.useContentCompression = false;
+        return this;
+    }
+
+    public HttpClientBuilder insecure(boolean insecure) {
+        allowAnyHostname(insecure);
+        allowAnyCertificate(insecure);
+        return this;
+    }
+
+    public HttpClientBuilder maxConnections(int maxConnections) {
+        this.maxConnections = maxConnections;
+        return this;
+    }
+
+    public HttpClientBuilder setDefaultRequestConfig(RequestConfig requestConfig) {
+        this.requestConfig = requestConfig;
+        return this;
+    }
+
+    public HttpClientBuilder useSystemProperties() {
+        return useSystemProperties(true);
+    }
+
+    public HttpClientBuilder useSystemProperties(boolean useSystemProperties) {
+        this.useSystemProperties = useSystemProperties;
+        return this;
+    }
+
+    public CloseableHttpClient build() {
+        org.apache.http.impl.client.HttpClientBuilder httpClientBuilder =
+                createHttpClientBuilder();
+
+        boolean acceptAnyCertificate = false;
+        boolean acceptAnyHostname = false;
+
+        if (useSystemProperties) {
```

Comment:
Why do we need this flag (useSystemProperties) ? Is it not enough and less confusing to check if those system properties are passed? Or is there some business logic I am missing?